### PR TITLE
bootloader: introduce managed bootloader, implement for grub

### PR DESCRIPTION
**Pull request by bboozzoo**
Introduce managed bootloader that models the idea of a bootloader which has its
boot config assets managed by snapd.

### CODE REVIEW

File: bootloader/bootloader.go [hunk 1]
```diff
@@ -139,6 +139,12 @@ type ExtractedRunKernelImageBootloader interface {
 	DisableTryKernel() error
 }
 
+// ManagedBootloader can manage its own boot assets (typically boot config).
+type ManagedBootloader interface {
```
Comment (pedronis) [inline bootloader/bootloader.go:143]: ManagedAssetsBootloader?

File: bootloader/bootloader.go [hunk 2]
```diff
@@ -139,6 +139,12 @@ type ExtractedRunKernelImageBootloader interface {
...
+	// IsManaged returns true when the on disk boot assets are managed.
+	IsManaged() (bool, error)
```
Comment (pedronis) [inline bootloader/bootloader.go:145]: for a second I wonder why the error, then I see this is a dynamic query about what is on disk? so maybe IsCurrentlyManaged ?

File: bootloader/grub.go [hunk 3]
```diff
@@ -300,3 +301,15 @@ func (g *grub) TryKernel() (snap.PlaceInfo, error) {
 	}
 	return nil, ErrNoTryKernelRef
 }
+
+// IsCurrentlyManaged returns true when the boot config is managed by snapd.
+//
+// Implements ManagedBootloader for the grub bootloader.
+func (g *grub) IsCurrentlyManaged() (bool, error) {
```
Comment (anonymouse64) [inline bootloader/grub.go:308]: do we need to distinguish between ubuntu-boot managed and ubuntu-seed managed in which case this should take a bootloader.Options too ?

Comment (bboozzoo) [reply bootloader/grub.go:308]: No, I don't think so.

**Pull request merged**